### PR TITLE
Move Commands into their own Classes

### PR DESCRIPTION
**Pull request by hsyyid**
**Task List:**
- [x] Move `CommandExecutors` into separate classes.
- [x] Register `SubCommands` correctly.

### CODE REVIEW

File: src/me/ryanhamshire/GriefPrevention/command/CommandDeleteClaim.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import org.spongepowered.api.entity.living.player.Player;
+
+public class CommandDeleteClaim implements CommandExecutor {
+
+    @Override
+    public CommandResult execute(CommandSource src, CommandContext ctx) {
+        Player player;
+        try {
+            player = GriefPrevention.instance.checkPlayer(src);
+        } catch (CommandException e) {
```
Comment (zml2008) [inline src/me/ryanhamshire/GriefPrevention/command/CommandDeleteClaim.java]: CommandExecutor throws CommandException -- why are you doing this?

File: src/me/ryanhamshire/GriefPrevention/command/CommandGriefPrevention.java [hunk 2]
```diff
@@ -63,11 +48,12 @@
...
     public static CommandSpec getCommand() {
         final ChildCommandElementExecutor commandExecutor = new ChildCommandElementExecutor(null);
-        commandExecutor.register(getAbandonAllClaimsCommand(), "abandonallclaims");
-        commandExecutor.register(getAbandonClaimCommand(), "abandonclaim", "unclaim", "declaim", "removeclaim", "disclaim");
-        commandExecutor.register(getAbandonTopLevelClaimCommand(), "abandontoplevelclaim");
-        commandExecutor.register(getIgnoreClaimsCommand(), "ignoreclaims", "ic");
-        commandExecutor.register(getFlagsCommand(), "flag");
+        HashMap<List<String>, CommandSpec> subcommands = GriefPrevention.instance.registerSubCommands();
+
+        for (List<String> aliases : subcommands.keySet()) {
```
Comment (zml2008) [inline src/me/ryanhamshire/GriefPrevention/command/CommandGriefPrevention.java]: Map.entrySet() is a thing

File: src/me/ryanhamshire/GriefPrevention/command/CommandGpBlockInfo.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import org.spongepowered.api.text.Texts;
+
+public class CommandGpBlockInfo implements CommandExecutor {
+
+    @Override
+    public CommandResult execute(CommandSource src, CommandContext ctx) {
+        Player player;
+
+        try {
+            player = GriefPrevention.instance.checkPlayer(src);
```
Comment (zml2008) [inline src/me/ryanhamshire/GriefPrevention/command/CommandGpBlockInfo.java]: Also checkPlayer should probably be static so you don't need to mess with the plugin instance here.

File: src/me/ryanhamshire/GriefPrevention/command/CommandAbandonClaim.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            player = GriefPrevention.instance.checkPlayer(src);
+        } catch (CommandException e) {
+            src.sendMessage(e.getText());
+            return CommandResult.success();
+        }
+        PlayerData playerData = GriefPrevention.instance.dataStore.getPlayerData(player.getUniqueId());
+
+        // which claim is being abandoned?
+        Claim claim = GriefPrevention.instance.dataStore.getClaimAt(player.getLocation(),
+                true /*
```
Comment (phroa) [inline src/me/ryanhamshire/GriefPrevention/command/CommandAbandonClaim.java]: I can't say I've ever seen this formatting before.

File: src/me/ryanhamshire/GriefPrevention/command/CommandAccessTrust.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import org.spongepowered.api.text.Texts;
+
+public class CommandAccessTrust implements CommandExecutor {
+
+    @Override
+    public CommandResult execute(CommandSource src, CommandContext ctx) {
+        try {
+            CommandHelper.handleTrustCommand(GriefPrevention.instance.checkPlayer(src), ClaimPermission.Access, ctx.<String>getOne("target").get());
+        } catch (CommandException e) {
+            src.sendMessage(Texts.of("An error occurred while executing this command."));
```
Comment (phroa) [inline src/me/ryanhamshire/GriefPrevention/command/CommandAccessTrust.java]: Doesn't Sponge handle this exact scenario?

File: src/me/ryanhamshire/GriefPrevention/command/CommandHelper.java [hunk 6]
```diff
@@ -102,4 +111,153 @@ else if (claim.children.size() > 0 && !deleteTopLevelClaim) {
...
+        UUID recipientID = null;
+        if (recipientName.startsWith("[") && recipientName.endsWith("]")) {
+            permission = recipientName.substring(1, recipientName.length() - 1);
+            if (permission == null || permission.isEmpty()) {
+                GriefPrevention.sendMessage(player, TextMode.Err, Messages.InvalidPermissionID);
+                return;
+            }
+        }
+
+        else if (recipientName.contains(".")) {
```
Comment (phroa) [inline src/me/ryanhamshire/GriefPrevention/command/CommandHelper.java]: I think your `else`s need to cozy up to the ending braces.

File: src/me/ryanhamshire/GriefPrevention/command/CommandBuyClaimBlocks.java [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+import org.spongepowered.api.command.CommandResult;
+import org.spongepowered.api.command.CommandSource;
+import org.spongepowered.api.command.args.CommandContext;
+import org.spongepowered.api.command.spec.CommandExecutor;
+
+public class CommandBuyClaimBlocks implements CommandExecutor {
+
+    @Override
+    public CommandResult execute(CommandSource src, CommandContext ctx) {
+        // TODO: Implement economy
```
Comment (phroa) [inline src/me/ryanhamshire/GriefPrevention/command/CommandBuyClaimBlocks.java]: why even have this?

**Comment by hsyyid** [comment]
@phroa For when Economy is implemented, nobody has to go and make the whole command executor class - also this commented out code was already there, I just moved it to a separate class.

File: src/me/ryanhamshire/GriefPrevention/command/CommandBuyClaimBlocks.java (hunk 7 above)
Comment (hsyyid) [inline src/me/ryanhamshire/GriefPrevention/command/CommandBuyClaimBlocks.java]: @phroa For when Economy is implemented, nobody has to go and make the whole command executor class - also this commented out code was already there, I just moved it to a separate class.

**Pull request merged**